src/riak_kv_pb_object.erl

```diff
@@ -190,8 +193,8 @@ process(#rpbputreq{bucket=B0, type=T, key=K, vclock=PbVC,
     case Result of
         consistent ->
             process(Req#rpbputreq{if_not_modified=undefined,
-                                  if_none_match=consistent},
-                    State);
+                                  if_none_match=undefined},
```

Comment:
Yeah, I agree it's not pretty. Still an improvement from the way it was though IMO, and I wanted to fix one hack without getting sucked into fixing all the other ugliness in this function. Trouble is, we need it to hit the next `process` clause after recursing, and if we leave `if_none_match` as true, it's going to loop back into the same clause again, so I think this is the best we can do without embarking on a big refactoring project.